src/microbe_stage/editor/CellEditorComponent.GUI.cs

```diff
@@ -302,14 +312,101 @@ private void UpdateOrganelleUnlockTooltips()
     {
-        // Don't use placeablePartSelectionElements as the thermoplast isn't placeable yet but is LAWK-dependent
         foreach (var entry in allPartSelectionElements)
         {
-            entry.Value.Visible = !Editor.CurrentGame.GameWorld.WorldSettings.LAWK || entry.Key.LAWK;
+            if (Editor.CurrentGame.GameWorld.WorldSettings.LAWK && !entry.Key.LAWK)
+                entry.Value.Hide();
+        }
+    }
+
+    private void CreateUndiscoveredOrganellesButtons(bool refresh = false, bool autoUnlock = true)
+    {
+        // Find groups with undiscovered organelles
+        var groupsWithUndiscoveredOrganelles =
+            new Dictionary<OrganelleDefinition.OrganelleGroup, (LocalizedStringBuilder UnlockText, int Count)>();
+
+        foreach (var entry in allPartSelectionElements)
+        {
+            var organelle = entry.Key;
+            var control = entry.Value;
+
+            // Skip already unlocked organelles
+            if (Editor.CurrentGame.GameWorld.UnlockProgress.IsUnlocked(organelle, Editor.CurrentGame, autoUnlock))
+            {
+                control.Undiscovered = false;
+                continue;
+            }
+
+            // Skip hidden organelles unless they are hidden because of missing requirements
+            if (!control.Visible && !control.Undiscovered)
+                continue;
+
+            control.Hide();
+            control.Undiscovered = true;
+
+            var buttonGroup = organelle.EditorButtonGroup;
+            var unlockTextString = new LocalizedString("UNLOCK_WITH_EITHER_OF_FOLLOWING", organelle.Name);
+
+            // Create unlock text
+            if (groupsWithUndiscoveredOrganelles.TryGetValue(buttonGroup, out var group))
+            {
+                // Add a new organelle to the group
+                group.Count += 1;
+                group.UnlockText.Append("\n\n");
+                group.UnlockText.Append(unlockTextString);
+                group.UnlockText.Append(" ");
+                organelle.GenerateUnlockRequirementsText(group.UnlockText);
+                groupsWithUndiscoveredOrganelles[buttonGroup] = group;
+            }
+            else
+            {
+                // Add the first organelle to the group
+                var unlockText = new LocalizedStringBuilder();
+                unlockText.Append(unlockTextString);
+                unlockText.Append(" ");
+                organelle.GenerateUnlockRequirementsText(unlockText);
+                groupsWithUndiscoveredOrganelles.Add(buttonGroup, (unlockText, 1));
+            }
+        }
+
+        // Remove any buttons that might've been created before
+        if (refresh)
+            RemoveUndisoveredOrganelleButtons();
+
+        // Generate undiscovered organelle buttons
+        foreach (var groupWithUndiscovered in groupsWithUndiscoveredOrganelles)
+        {
+            var group = partsSelectionContainer.GetNode<CollapsibleList>(groupWithUndiscovered.Key.ToString());
```

Comment:
This seems a bit brittle as this requires a specific name node. Though, I guess this might be exactly what's done elsewhere in the editor (I can't immediately remember), if this is done elsewhere in the editor then this is fine and not any more problematic than other editor code.

Reply:
Yup it appears that's how it's used in the other parts of the editor code